packages/cxl-ui/src/components/cxl-star-rating.js

```diff
@@ -0,0 +1,40 @@
+import '@conversionxl/cxl-lumo-styles';
+import { IronStarRating } from '@cwmr/iron-star-rating';
+
+export class CXLStarRating extends IronStarRating {
+  static get is() {
+    return 'cxl-star-rating';
+  }
+
+  _getUniqueId() {
+    return `cxl-star-rating_${this.id}`;
```

Comment:
Is `id` guaranteed to be set always?

Reply:
It puts `cxl-star-rating_undefined` in case `id` is not defined, but the user should **always** provide `id` attribute in order to distinct ratings across playbooks. IMO documentation should inform the user about the necessity of providing `id` here unless you have another idea to force the user to provide the unique identifier for rating?

Reply:
> IMO documentation should inform the user about the necessity of providing id here unless you have another idea to force the user to provide the unique identifier for rating?

In HTML, it's unfeasible to enforce anything.

I think the code should just not store anything if no valid identifier provided.

Fairly certainly cxl-vaadin-accordion operates like this.

Reply:
Won't the user be confused why the double rating is **blocked** only if `id` isn't provided? Shouldn't this be an additional attribute, with a more `self-explanatory` name like: `vote-once=true`? If the user does not provide `id` we could make a unique `localStorage` identifier with `encodeURIComponent(window.location.pathname)`.

Reply:
There's no need to overcomplicate matters. Our "users" is a very small circle of devs, we're not targeting this for general public.

Maybe can just document "`id` is required for state storage" in JSDoc for better visibility.

Reply:
OK, I've changed it here: https://github.com/conversionxl/aybolit/pull/88/commits/59cd81972196f46b0bc051985b2d4426fed7a64a